MvvmCross-Forms/MvvmCross.Forms/Views/MvxFormsPagePresenter.cs

```diff
@@ -150,34 +151,79 @@ public override void RegisterAttributeTypes()
                 });
         }
 
-        public override void ChangePresentation(MvxPresentationHint hint)
+        public async override void ChangePresentation(MvxPresentationHint hint)
         {
             var navigation = GetPageOfType<NavigationPage>().Navigation;
             if (hint is MvxPopToRootPresentationHint popToRootHint)
             {
-                navigation.PopToRootAsync(popToRootHint.Animated);
+                // Make sure the modals are all closed
+                while (navigation.ModalStack.Any())
+                {
+                    await navigation.PopModalAsync();
+                }
+
+                await navigation.PopToRootAsync(popToRootHint.Animated);
                 return;
             }
             if (hint is MvxPopPresentationHint popHint)
             {
-                foreach (var page in navigation.NavigationStack)
+                // Need to check the modal stack first
+                while (navigation.ModalStack.Any())
```

Comment:
This should go into `CloseAllModals` or another separate method to prevent it being so big.

Reply:
Yep, was going to refactor this section

Reply:
@martijn00  I think the code is a lot cleaner now - let me know what you think

Reply:
The normal popping is indeed, but you can move this modal section into a method too i think.